kobo/worker/logger.py

```diff
@@ -71,6 +71,10 @@ def stop(self):
 
 class LoggingIO(object):
     """StringIO wrapper that also writes all data to a logging thread."""
+    __slots__ = (
```

Comment:
Can you please explain why the absence of these slots causes the output interleaving mentioned in the commit message?

Reply:
Thanks, I was able to reproduce it.

It turns out that this is a bug in kobo.shortcuts.run, which is reproducible standalone without using other parts of kobo.  I've filed it as https://github.com/release-engineering/kobo/issues/40.

In the covscan case having the `__slots__` on LoggingIO will effectively hide the bug.  The reason is because of the way print statement works - per https://docs.python.org/2/reference/simple_stmts.html#print:

>  A space is written before each object is (converted and) written, unless the output system believes it is positioned at the beginning of a line.

The state "positioned at the beginning of a line" is maintained by stashing a softspace attribute on the file object.  When `__slots__` are present on LoggingIO, and not containing softspace, the runtime will silently fail to store that attribute on the file object, so will perform all prints without additional whitespace.  When `__slots__` are removed then the softspace attribute can be stored, so prints get leading whitespace per the documented behavior, exposing the bug in run method.

I think you also wouldn't notice this bug in covscan if you weren't using buffer_size=2 to kobo run (is that intentional by the way?  Seems an oddly small buffer.)

I don't really want to get into fixing run right now so I'm fine with merging this, but do you think you could tweak the commit message to have a proper explanation?  i.e. mention that it's a workaround for https://github.com/release-engineering/kobo/issues/40 .